src/components/MAPI/apps/AppDetailsModal/index.tsx

```diff
@@ -167,7 +167,7 @@ const AppDetailsModal: React.FC<IAppDetailsModalProps> = ({
       );
 
       mutateApp(updatedApp);
-      // TODO(axbarsan): Mutate apps list, once that request exists.
+      mutate(applicationv1alpha1.getAppListKey({ namespace: clusterId }));
```

Comment:
This makes sure that we re-execute the app fetching request after we delete an app, to get the updated list of apps